### ISSUE TITLE
Is this library still working?

### ISSUE THREAD

**Issue by achapela** [BLavery/lib_nrf24#5]
I cannot make this library work anymore. On the receiving end, the "available" function always returns true, even if no signal is being received. 

Is this library still working? Or is there another one that I could try out?
Please help.

**Comment by ChienDesEnfers** [comment]
Hello @achapela ,
Did you find a way to use this library?
I'm trying to use it to send/receive data on my Raspberry 2B but I have the same issue: the available function always return True =/
Thank you for your help

**Comment by Abi1506** [comment]
Yes. It´s working, I'm using in Raspberry PI 3 with Raspbian Jessy.

**Comment by ChienDesEnfers** [comment]
@Abi1506 Thank you for your answer :) I finally find another library to make it works.

**Comment by Gabedu2000** [comment]
@ChienDesEnfers Hi What library work? Thanks

**Comment by ChienDesEnfers** [comment]
@Gabedu2000 I use[ this one](https://github.com/ChienDesEnfers/RF24) on my repo.
I just finished to create a readme file to explain steps to install it (with Python).
Don't hesitate to let me a comment if there are some mistake into my readme.

**Comment by gcenjor** [comment]
**No, actually doesn't work.** 
Something changed in the Raspberry after an update.

Read this thread:
[https://raspberrypi.stackexchange.com/questions/71730/possible-spi-issues-around-nrf24-operations-post-rpi-system-update](url)

It´s necessary to add 
`self.spidev.max_speed_hz = 100000`
in lib_nrf24.py

In the post say 4.000.000 but in my case I had to lower to 100.000 to make it works.

**Comment by achapela** [comment]
Thank you all for the help. I didn't make it work so I moved to another library. Cheers

**Comment by fcanozkan** [comment]
@gcenjor Your fix worked  thanks! 

But your link no longer works here is the working link: https://raspberrypi.stackexchange.com/questions/71730/possible-spi-issues-around-nrf24-operations-post-rpi-system-update

Also for future referance you need to change in lib_nrf24.py in function:

def begin(self, csn_pin ... after  self.spidev.open add:

self.spidev.max_speed_hz = 100000

**Comment by raspbuino** [comment]
**@achapela** thank you for opening this thread! I had exactly the same issue after an update to Raspbian Stretch and five minutes ago I finally managed to find this thread by the help of the "big G"
**@fcanozkan** and **@gcenjor** thank you for describing the solution - it works for me too.
You made my day!

**Comment by markandkymward** [comment]
This library no longer works on Buster and RPi 4.  I've tried every possible combination of CE, CSN attachments and did the "self.spidev.max_speed_hz = " hack in lib_nrf24.py using various values.  Can't seem to get it to talk to the radio.  Also tried several different radios.  Also - I have to radios humming away on two different arduinos sitting next to it so I know there is valid data being transmitted.  Any additional help is very welcome!

**Comment by Soneca-ii** [comment]
using the hack 
self.spidev.max_speed_hz = 100000

I can confirm that it worked in a raspberry pi zero w

somehow I needed to do a reboot after multiple tries and it works

**Comment by zetoune2** [comment]
I confirm what @markandkymward mentioned:

Despite numerous attempt I can not make it work on a raspberryPi4. The Pi refuse to send data. Also down same test with 2 Arduino and the rf24 transmitter/receiver are working fine. 

Will check if other library make it work but for the moment I am stuck. Any help you may have would be appreciated

 > This library no longer works on Buster and RPi 4. I've tried every possible combination of CE, CSN attachments and did the "self.spidev.max_speed_hz = " hack in lib_nrf24.py using various values. Can't seem to get it to talk to the radio. Also tried several different radios. Also - I have to radios humming away on two different arduinos sitting next to it so I know there is valid data being transmitted. Any additional help is very welcome!
